apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
-      });
-    } else {
-      callbackify(copyQuestionBetweenCourses)(
-        res,
-        {
+        editor.assertCanEdit();
+        await editor.doEditAsync();
+        const questionId = await sqldb.queryRow(
+          sql.select_question_id_from_uuid,
+          { uuid: editor.uuid, course_id: res.locals.course.id },
+          IdSchema,
+        );
+        flash(
+          'success',
+          'Question copied successfully. You are now viewing your copy of the question.',
+        );
+        res.redirect(res.locals.urlPrefix + '/question/' + questionId + '/settings');
+      } else {
+        copyQuestionBetweenCourses(res, {
           fromCourse: res.locals.course,
           toCourseId: req.body.to_course_id,
           question: res.locals.question,
-        },
-        (err) => {
-          if (ERR(err, next)) return;
-          // `copyQuestionBetweenCourses` performs the redirect automatically,
-          // so if there wasn't an error, there's nothing to do here.
-        },
-      );
-    }
-  } else if (req.body.__action === 'delete_question') {
-    debug('Delete question');
-    const editor = new QuestionDeleteEditor({
-      locals: res.locals,
-    });
-    editor.canEdit((err) => {
-      if (ERR(err, next)) return;
-      editor.doEdit((err, job_sequence_id) => {
-        if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-          res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-        } else {
-          res.redirect(res.locals.urlPrefix + '/course_admin/questions');
-        }
+        });
+      }
+    } else if (req.body.__action === 'delete_question') {
+      const editor = new QuestionDeleteEditor({
+        locals: res.locals,
       });
-    });
-  } else if (req.body.__action === 'sharing_set_add') {
-    debug('Add question to sharing set');
-    features
-      .enabledFromLocals('question-sharing', res.locals)
-      .then((questionSharingEnabled) => {
-        if (!questionSharingEnabled) {
-          return next(error.make(403, 'Access denied (feature not available)'));
-        }
-        if (!res.locals.authz_data.has_course_permission_own) {
-          return next(error.make(403, 'Access denied (must be a course Owner)'));
-        }
-        sqldb.queryZeroOrOneRow(
-          sql.sharing_set_add,
-          {
+      editor.assertCanEdit();
+      await editor.doEditAsync();
+      res.redirect(res.locals.urlPrefix + '/course_admin/questions');
+    } else if (req.body.__action === 'sharing_set_add') {
+      features
+        .enabledFromLocals('question-sharing', res.locals)
+        .then((questionSharingEnabled) => {
+          if (!questionSharingEnabled) {
+            throw error.make(403, 'Access denied (feature not available)');
+          }
+          if (!res.locals.authz_data.has_course_permission_own) {
+            throw error.make(403, 'Access denied (must be a course Owner)');
+          }
+          sqldb.queryAsync(sql.sharing_set_add, {
```

Comment:
You need to await async functions.

```suggestion
          await sqldb.queryAsync(sql.sharing_set_add, {
```

Reply:
Updated locally in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).